src/OpenSpaceToolkit/Astrodynamics/Trajectory/Sequence.cpp

```diff
@@ -222,17 +222,30 @@ void Sequence::addManeuverSegment(const Shared<EventCondition>& anEventCondition
     segments_.add(Segment::Maneuver("Maneuver", anEventConditionSPtr, aThruster, dynamics_, numericalSolver_));
 }
 
-Sequence::Solution Sequence::solve(const State& aState) const
+Sequence::Solution Sequence::solve(
+    const State& aState, const Duration& aMaximumPropagationDuration, const Shared<EventCondition>& anEventConditionSPtr
+) const
 {
     Array<Segment::Solution> segmentSolutions;
 
     State initialState = aState;
     State finalState = State::Undefined();
 
+    bool eventConditionIsSatisfied = anEventConditionSPtr == nullptr;
+
     for (Size i = 0; i < repetitionCount_; ++i)
     {
         for (const Segment& segment : segments_)
         {
+            // Maximum propagation check
+            if ((initialState.accessInstant() - aState.accessInstant()) > aMaximumPropagationDuration)
+            {
+                BOOST_LOG_TRIVIAL(warning)
+                    << "Terminating Sequence because maximum propagation duration is reached." << std::endl;
```

Comment:
```suggestion
                    << "Terminating Sequence because maximum propagation duration was reached in the preceding segment." << std::endl;
```

nit: just thinking about that when this happens, the last segment will likely have gone _past_ the maximum duration. Might be good to indicate somehow that this check is discretized by segments to avoid confusion.

In general, I could see this condition being a little confusing. For example, if someone is testing a low-thrust maneuver to get to some altitude that they may never reach, the intuitive thing to do is probably to set this `maximumPropagationDuration`, so that it doesn't go forever. However, I think this condition alone would not be sufficient. They would need a maximum duration condition on the maneuver segment as well (or a maximum fuel used or something). In that case, this condition would appear to do nothing.

Let me know if I'm misunderstanding here. Perhaps we could name the condition something slightly different so it's clear that it's only applied at discrete instants between segments.

Reply:
yeah I think this is the confusion. We have 2 sets of `maximumPropagationDurations`. There's one for a Segment, so that any individual Segment doesn't blast off into infinity.
And then there's the one for a Sequence. So that if you have like 20 segments, each with a max of 7 days, you might still not want to run the sequence for more than 30 days or something.

I can update the naming so that the Sequence constructor has the `maximumPropagationDuration` changed to `segmentPropagationDurationLimit` and then the one in this function can remain. Is that better? @kyle-cochran

Reply:
Hmmm... I think for the the `maximumPropagationDuration` naming implies a condition on "continuous" duration rather than duration discretized by segments. What about calling the sequence one something like: `maximumSegmentsDuration`. Trying to imply something like: "the sum of all of the segments cannot exceed X"

?